lib/backend/etcd/etcd.go

```diff
@@ -120,6 +121,35 @@ func (c *EtcdClient) EnsureInitialized() error {
 		log.Info("Ready flag is already set")
 	}
 
+	// The confd agent in calico/node requires a few directories to exist
+	// to be able to monitor correctly:
+	// -  IP pools parent directories
+	// -  BGP custom filters directories.
+	if err := c.ensureDirectory("/calico/v1/ipam/v4/pool"); err != nil {
+		return err
+	} else if err = c.ensureDirectory("/calico/v1/ipam/v6/pool"); err != nil {
+		return err
+	} else if err = c.ensureDirectory("/calico/bgp/v1/global/custom_filters/v4"); err != nil {
+		return err
+	} else if err = c.ensureDirectory("/calico/bgp/v1/global/custom_filters/v6"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// EnsureInitialized makes sure that the etcd data is initialized for use by
+// Calico.
+func (c *EtcdClient) EnsureNodeInitialized(node string) error {
+	// The confd agent in calico/node requires the IPAM affinity directories
```

Comment:
confd isn't just for calico/node - confd will need this even when running outside of a container.